lab_4_retrieval_w_clustering/main.py

```diff
@@ -113,6 +152,23 @@ def put_corpus(self, corpus: Corpus) -> None:
                 or if input arguments are empty,
                 or if methods used return None.
         """
+        if not isinstance(corpus, list) or not corpus:
+            raise ValueError("Input arguments are wrong")
+        tokenized_documents = self._tokenizer.tokenize_documents(corpus)
+        if tokenized_documents is None:
+            raise ValueError("Tokenization returned None")
+
+        filtered_documents = [self._tokenizer._remove_stop_words(paragraph) for paragraph in
```

Comment:
Unfortunately, it is prohibited to use protected functions outside class implementation, so here you should use `self._tokenizer.tokenize()` function

Reply:
done